Assets/Scripts/Source/Game/GameFlowManager.cs

```diff
@@ -48,10 +48,38 @@ public Enemy GetEnemy()
     }
 }
 
+[Serializable]
+public class WeaponSelect
```

Comment:
Nit: have we considered making `Weapon` serializable instead?

Reply:
I feel like their was an attempt to do that but I didn't. :thinking: